app/services/sign_in/service_account_access_token_jwt_encoder.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module SignIn
+  class ServiceAccountAccessTokenJwtEncoder
+    attr_reader :decoded_service_account_assertion
+
+    def initialize(decoded_service_account_assertion:)
+      @decoded_service_account_assertion = decoded_service_account_assertion
+    end
+
+    def perform
+      service_account_access_token
+    end
+
+    private
+
+    def service_account_access_token
+      @service_account_access_token ||= jwt_encode_access_token
+    end
+
+    def jwt_encode_access_token
+      JWT.encode(payload, private_key, Constants::ServiceAccountAccessToken::JWT_ENCODE_ALGORITHM)
+    end
+
+    def payload
+      {
+        iss: "#{hostname}/#{Constants::ServiceAccountAccessToken::ISSUER}",
```

Comment:
Is hostname needed here?

Reply:
My interpretation of `iss` is that it's supposed to identify the specific application creating the JWT, so dev vs. staging vs production is a distinction we'd want.